barretenberg/sol/src/honk/ZKTranscript.sol

```diff
@@ -0,0 +1,475 @@
+pragma solidity >=0.8.21;
+
+import {
+    Honk,
+    NUMBER_OF_ALPHAS,
+    NUMBER_OF_ENTITIES,
+    ZK_BATCHED_RELATION_PARTIAL_LENGTH,
+    CONST_PROOF_SIZE_LOG_N
+} from "./HonkTypes.sol";
+import {Fr, FrLib} from "./Fr.sol";
+
+import {bytesToG1ProofPoint, bytesToFr, logFr, logG} from "./utils.sol";
+
+import "forge-std/console.sol";
+
+// ZKTranscript library to generate fiat shamir challenges, the ZK transcript only differest
+struct ZKTranscript {
```

Comment:
Identical in structure with the non-ZK transcript, but we generate an additional challenge in sumcheck and we need to hash more data in places because the ZK proof includes extra field elements and commitments.